Tests/SourceKitLSPTests/ExecuteCommandTests.swift

```diff
@@ -141,120 +159,362 @@ final class ExecuteCommandTests: XCTestCase {
-      )
+          _ = try await project.testClient.send(request)
+
+          try await fulfillmentOfOrThrow([expectation])
+
+          let urls = try XCTUnwrap(
+            peekDocumentsRequestURIs.value?.map {
+              return try XCTUnwrap(
+                $0.fileURL,
+                "Failed for position range between \(positionMarker.start) and \(positionMarker.end)"
+              )
+            },
+            "Failed for position range between \(positionMarker.start) and \(positionMarker.end)"
+          )
+
+          let filesContents = try urls.map { try String(contentsOf: $0, encoding: .utf8) }
+
+          XCTAssertEqual(
+            filesContents,
+            [
+              "@DictionaryStorageProperty",
+              "@DictionaryStorageProperty",
+              "var _storage: [String: Any] = [:]",
+            ],
+            "Files doesn't contain correct macro expansion. Failed for position range between \(positionMarker.start) and \(positionMarker.end)"
+          )
+
+          XCTAssertEqual(
+            urls.map { $0.lastPathComponent }.sorted(),
+            [
+              "MyMacroClient_L11C3-L11C3.swift",
+              "MyMacroClient_L12C3-L12C3.swift",
+              "MyMacroClient_L13C1-L13C1.swift",
+            ].sorted(),
+            "Failed for position range between \(positionMarker.start) and \(positionMarker.end)"
+          )
+        } else {
+          let expectation = self.expectation(description: "Handle Show Document Request")
+          let showDocumentRequestURI = ThreadSafeBox<DocumentURI?>(initialValue: nil)
+
+          project.testClient.handleSingleRequest { (req: ShowDocumentRequest) in
+            showDocumentRequestURI.value = req.uri
+            expectation.fulfill()
+            return ShowDocumentResponse(success: true)
+          }
 
-      let fileContents = try String(contentsOf: url, encoding: .utf8)
+          _ = try await project.testClient.send(request)
 
-      XCTAssert(
-        fileContents.contains("(1 + 2, \"1 + 2\")"),
-        "File doesn't contain macro expansion. Failed for position range between \(positionMarker.start) and \(positionMarker.end)"
-      )
+          try await fulfillmentOfOrThrow([expectation])
 
-      XCTAssertEqual(
-        url.lastPathComponent,
-        "MyMacroClient_L4C2-L4C19.swift",
-        "Failed for position range between \(positionMarker.start) and \(positionMarker.end)"
-      )
+          let url = try XCTUnwrap(
+            showDocumentRequestURI.value?.fileURL,
+            "Failed for position range between \(positionMarker.start) and \(positionMarker.end)"
+          )
+
+          let fileContents = try String(contentsOf: url, encoding: .utf8)
+
+          XCTAssertEqual(
+            fileContents,
+            "// MyMacroClient.swift @ 11:3 - 11:3\n@DictionaryStorageProperty\n// MyMacroClient.swift @ 12:3 - 12:3\n@DictionaryStorageProperty\n// MyMacroClient.swift @ 13:1 - 13:1\nvar _storage: [String: Any] = [:]",
```

Comment:
Let’s use a multi-line string literal here. I think that’s easier to understand as a human.

Also, I think it would be nice to have an empty line before every `//` comment that’s not on the first line in the file, it loosens up the file a little. Ie. something like

```suggestion
            """
            // MyMacroClient.swift @ 11:3 - 11:3
            @DictionaryStorageProperty
            
            // MyMacroClient.swift @ 12:3 - 12:3
            @DictionaryStorageProperty

            // MyMacroClient.swift @ 13:1 - 13:1
            var _storage: [String: Any] = [:]",
```

Reply:
Or should I allow this to be:

```
// MyMacroClient.swift @ 11:3 - 11:3
@DictionaryStorageProperty
// MyMacroClient.swift @ 12:3 - 12:3
@DictionaryStorageProperty
// MyMacroClient.swift @ 13:1 - 13:1
var _storage: [String: Any] = [:]",
            
```

Do you want me to let the macro expansion implementation to add new lines before the comments?

Reply:
I’m not entirely sure what you mean.